src/dlaf_fortran.f90

```diff
@@ -110,12 +139,22 @@ end subroutine dlaf_free_grid_c
    end subroutine dlaf_free_grid
 
    subroutine dlaf_pspotrf(uplo, n, a, ia, ja, desca, info)
+      !! Cholesky decomposition for a distributed single-precision real symmetric positive definite matrix \(\mathbf{A}\)
+      !! {!docs/snippets/note-host-matrix.md!}
       character, intent(in) :: uplo
+      !! {!docs/snippets/uplo.md!}
       integer, intent(in) :: n
+      !! {!docs/snippets/n.md!}
       real(kind=sp), dimension(:, :), target, intent(inout) :: a
-      integer, intent(in) :: ia, ja
+      !! {!docs/snippets/a.md!}
+      integer, intent(in) :: ia
+      !! {!docs/snippets/ia.md!}
+      integer, intent(in) :: ja
+      !! {!docs/snippets/ja.md!}
       integer, dimension(9), intent(in) :: desca
+      !! {!docs/snippets/desca.md!}
       integer, target, intent(out) :: info
+      !! {!docs/snippets/info.md!}
```

Comment:
I admit I didn't expect to see this here, but it's really nice (I'm referring to the inclusion of snippets)! My only concern is that this means there are no (readable) docs just while browsing the code, but I think it's still a net benefit to ensure the published documentation is consistent. Most users will not be browsing the code.

Reply:
I too thought that it a net benefit. (You have to look only at a few files to spot all my typos... =P)